### PR TITLE
Refactor liveness client

### PR DESCRIPTION
**Pull request by maciekzygmunt**
This PR adds small refactoring to `LivenessIndexer` and `LivenessClient`.

### CODE REVIEW

File: packages/backend/src/core/liveness/LivenessIndexer.ts [hunk 1]
```diff
@@ -40,30 +45,35 @@ export class LivenessIndexer extends ChildIndexer {
...
-            id,
+            c.livenessConfigurationId,
             adjustedTo,
             trx,
           ),
         ),
       )
     })
 
+    const usedConfigurationsIds = [
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessIndexer.ts:72]: there is no need to map, please return transfer configs length and function configs length, so we will more granularily see what happened

File: packages/backend/src/core/liveness/LivenessIndexer.ts [hunk 2]
```diff
@@ -40,30 +45,35 @@ export class LivenessIndexer extends ChildIndexer {
...
-      await this.livenessClient.getLivenessData(
-        this.projects,
-        configurations,
-        new UnixTime(from),
-        new UnixTime(to),
-      )
+    const { transfersConfig, functionCallsConfig, adjustedTo } =
+      await this.getConfiguration(from, to)
+
+    const data = await this.livenessClient.getLivenessData(
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessIndexer.ts:51]: if both configurations are empty do not call liveness client, instead just log that "Update skipped" {from, adjustedTo}

File: packages/backend/src/core/liveness/LivenessIndexer.ts [hunk 3]
```diff
@@ -40,30 +45,35 @@ export class LivenessIndexer extends ChildIndexer {
...
         ),
       )
     })
 
+    const usedConfigurationsIds = [
+      ...transfersConfig.map((c) => c.livenessConfigurationId),
+      ...functionCallsConfig.map((c) => c.livenessConfigurationId),
+    ]
+
     this.logger.info('Updated liveness data', {
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessIndexer.ts:77]:
```suggestion
    this.logger.info('Updated', {
```

File: packages/backend/src/core/liveness/LivenessClient.test.ts [hunk 4]
```diff
@@ -28,13 +29,35 @@ describe(LivenessClient.name, () => {
...
 
-      const { data, adjustedTo, usedConfigurationsIds } =
-        await livenessClient.getLivenessData(
-          PROJECTS,
-          LIVENESS_CONFIGURATIONS,
+      const config = mergeConfigs(PROJECTS, LIVENESS_CONFIGURATIONS)
+
+      const adjustedTo = adjustToForBigqueryCall(FROM.toNumber(), TO.toNumber())
+
+      const transfersConfig = config.transfers.filter((c) =>
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessClient.test.ts:36]: there is no need to filter there

File: packages/backend/src/core/liveness/LivenessClient.test.ts [hunk 5]
```diff
@@ -28,13 +29,35 @@ describe(LivenessClient.name, () => {
 
       const livenessClient = new LivenessClient(bigquery)
 
-      const { data, adjustedTo, usedConfigurationsIds } =
-        await livenessClient.getLivenessData(
-          PROJECTS,
-          LIVENESS_CONFIGURATIONS,
+      const config = mergeConfigs(PROJECTS, LIVENESS_CONFIGURATIONS)
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessClient.test.ts:32]: extract this to MERGED_CONFIGS

File: packages/backend/src/core/liveness/LivenessClient.test.ts [hunk 6]
```diff
@@ -28,13 +29,35 @@ describe(LivenessClient.name, () => {
...
+
+      const adjustedTo = adjustToForBigqueryCall(FROM.toNumber(), TO.toNumber())
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessClient.test.ts:34]: There is no need to adjsut

File: packages/backend/src/core/liveness/LivenessIndexer.ts [hunk 7]
```diff
@@ -40,39 +46,85 @@ export class LivenessIndexer extends ChildIndexer {
...
+            c.livenessConfigurationId,
             adjustedTo,
             trx,
           ),
         ),
       )
     })
 
-    this.logger.info('Updated liveness data', {
+    const usedConfigurationsLength =
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessIndexer.ts:78]: inline this value in return

File: packages/backend/src/core/liveness/LivenessIndexer.ts [hunk 8]
```diff
@@ -40,39 +46,85 @@ export class LivenessIndexer extends ChildIndexer {
...
+      await this.getConfiguration(from, to)
 
-    const { data, adjustedTo, usedConfigurationsIds } =
-      await this.livenessClient.getLivenessData(
-        this.projects,
-        configurations,
-        new UnixTime(from),
-        new UnixTime(to),
-      )
+    if (transfersConfig.length === 0 && functionCallsConfig.length === 0) {
```
Comment (antooni) [inline packages/backend/src/core/liveness/LivenessIndexer.ts:52]: add a test for this

File: packages/backend/src/core/liveness/utils/mergeConfigs.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import { assert } from '@l2beat/backend-tools'
+import { notUndefined } from '@l2beat/shared-pure'
+
+import { Project } from '../../../model'
+import { LivenessConfigurationRecord } from '../../../peripherals/database/LivenessConfigurationRepository'
+import { LivenessFunctionCall, LivenessTransfer } from '../types/LivenessConfig'
+import { LivenessConfigurationIdentifier } from '../types/LivenessConfigurationIdentifier'
+
+export function mergeConfigs(
```
Comment (sz-piotr) [inline packages/backend/src/core/liveness/utils/mergeConfigs.ts:9]: maybe this function should be called `getLivenessQueries`?

File: packages/backend/src/core/liveness/LivenessIndexer.test.ts [hunk 10]
```diff
@@ -349,4 +375,40 @@ describe(LivenessIndexer.name, () => {
       expect(value).toEqual(1)
     })
   })
+
+  describe(LivenessIndexer.prototype.getConfiguration.name, () => {
+    it('should return configurations and adjustedTo', async () => {
```
Comment (sz-piotr) [inline packages/backend/src/core/liveness/LivenessIndexer.test.ts:380]: This test doesn't really test as it just duplicates the implementation code

**Review by sz-piotr — DISMISSED** [review]

**Pull request merged**